File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        // Create a dummy node that points to the head of the list
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        
        // Initialize current and previous pointers
        ListNode current = head;
        ListNode prev = dummy;
        
        while (current != null) {
            if (current.val == val) {
                // Skip the current node
                prev.next = current.next;
            } else {
                // Move the prev pointer forward
                prev = current;
            }
            // Move the current pointer forward
            current = current.next;
        }
        
        // Return the new head of the list
        return dummy.next;
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy=ListNode()
        dummy.next=head
        current,prev=head,dummy
        while current:
            if current.val==val:
                prev.next=current.next
            else:
                prev=current
            current=current.next
        return dummy.next
